client/sites/marketing/tools/index.tsx

```diff
@@ -20,17 +24,46 @@ import './style.scss';
 export default function MarketingTools() {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
-	const recordTracksEvent = ( event: string ) => dispatch( recordTracksEventAction( event ) );
+	const [ searchTerm, setSearchTerm ] = useState( '' );
 	const selectedSiteSlug: T.SiteSlug | null = useSelector( getSelectedSiteSlug );
 	const siteId = useSelector( getSelectedSiteId ) || 0;
 
+	const recordTracksEvent = useCallback(
+		( event: string, properties?: Record< string, string > ) =>
+			dispatch( recordTracksEventAction( event, properties ) ),
+		[ dispatch ]
+	);
+
 	const marketingFeatures = getMarketingFeaturesData(
 		selectedSiteSlug,
 		recordTracksEvent,
 		translate,
 		localizeUrl
 	);
 
+	const marketingFeaturesFiltered = useMemo( () => {
+		return marketingFeatures.filter(
+			( feature ) =>
+				feature.title.toLowerCase().includes( searchTerm.toLowerCase() ) ||
+				feature.description.toLowerCase().includes( searchTerm.toLowerCase() )
+		);
+	}, [ searchTerm, marketingFeatures ] );
+
+	const handleRecordSearch = useMemo(
+		() =>
+			debounce(
+				( term ) =>
+					recordTracksEvent( `calypso_marketing_tools_business_tools_search`, {
+						search_term: term as string,
+					} ),
+				500
+			),
+		[ recordTracksEvent ]
+	);
+	useEffect( () => {
+		handleRecordSearch( searchTerm );
```

Comment:
We should add a condition for `searchTerm !== ''` so that we don't track empty searches, which happens - especially when you just load the page.

![CleanShot 2024-11-22 at 15 48 05](https://github.com/user-attachments/assets/8acffe6e-e06a-408c-9a25-44560ae44e65)

Reply:
Ok, I took into account deleting the search as something we would like to track but didn't notice the initial track being sent. I will add the condition so we save some unnecessary events. Well spotted!

Reply:
Condition added here a572191